pkg/pillar/cmd/zedrouter/ipaddronunet.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2021 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Network Instance underlay network Application Number Management
+
+package zedrouter
+
+import (
+	"fmt"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+)
+
+func appNumsOnUNetAllocate(ctx *zedrouterContext,
+	config *types.AppNetworkConfig) {
+	for _, ulConfig := range config.UnderlayNetworkList {
+		appID := config.UUIDandVersion.UUID
+		networkID := ulConfig.Network
+		netStatus := lookupNetworkInstanceStatus(ctx, networkID.String())
+		if netStatus != nil {
+			baseSize := netStatus.DhcpRange.Size
+			appNum, err := appNumOnUNetAllocate(ctx, networkID, appID,
+				baseSize, false)
+			if appNum == types.AppNumInvalid || err != nil {
+				errStr := fmt.Sprintf("App Num get fail :%s", err)
+				log.Errorf("appNumsOnAppNetworkAllocate(%s, %s): fail: %s\n",
+					networkID.String(), appID.String(), errStr)
+			}
+		}
+	}
+}
+
+func appNumsOnUNetFree(ctx *zedrouterContext,
+	status *types.AppNetworkStatus) {
+	appID := status.UUIDandVersion.UUID
+	for ulNum := 0; ulNum < len(status.UnderlayNetworkList); ulNum++ {
+		ulStatus := &status.UnderlayNetworkList[ulNum]
+		networkID := ulStatus.Network
+		// release the app number
+		_, err := appNumOnUNetGet(ctx, networkID, appID)
+		if err == nil {
+			appNumOnUNetFree(ctx, networkID, appID)
+		}
+	}
+}
+
+// Assignable IP Address counter routines,
+// for this network instance
+func getAppCountOnUNet(ctx *zedrouterContext,
+	netInstStatus *types.NetworkInstanceStatus) int {
+	hostCount := 0
+	if doNetworkInstanceStatusDhcpRangeSanityCheck(netInstStatus) == nil {
+		hostCount = netInstStatus.DhcpRange.HostCount()
+		subnetHostCount := types.GetIPAddrCountOnSubnet(netInstStatus.Subnet)
+		// if hostCount for dhcp range is equal to or, more than subnet size,
+		// deduct network, broadcast and gateway
+		// and restrict it to AppNumMax
```

Comment:
Why do you need this? You added code to adjust the DhcpRange to not include the subnet and broadcast address (and also verify that the gateway isn't in the range). Tith that addition the size of the dhcp range will indicate how many host addresses can fit.